### ISSUE TITLE
instance() on stateless components

### ISSUE THREAD

**Issue by laumair** [airbnb/enzyme#666]
Is there any way to test methods inside stateless components? I know it isn't recommended to load up methods inside stateless functional components but just curious to know if there's any way to test those functions?

**Comment by ljharb** [comment]
stateless components don't have methods, that's the point - can you show an example of what you mean?

In general, private things should not be tested, so if it's not exposed, you shouldn't be trying to test it directly.

**Comment by laumair** [comment]
@ljharb Yeah I get that stateless components don't have methods. What I meant was:

```
const Component = (props) => {
const filteredData = props.data => {}; // What if this is to be tested?
return (
<div>
{filteredData.map(...)}
</div>
);
};
export default Component;
```
This is just a basic example of what I was trying to say. In one or two places I ended up having complex logic as functions inside stateless components that needed to be tested but I couldn't so I had to instead use a class and reading around I got to know we SHOULD NOT be using stateless functional components if we have such logic inside our components. But, this is just for curiosity if we were to test something like this directly?
